modules/gateways/monero/createinvoice.php

```diff
@@ -34,8 +34,9 @@ function monero_payment_id(){
 $currency = stripslashes($_POST['currency']);
 $amount_xmr = stripslashes($_POST['amount_xmr']);
 $amount = stripslashes($_POST['amount']);
-$payment_id = monero_payment_id();
 $invoice_id = stripslashes($_POST['invoice_id']);
+$client_id = stripslashes($_POST['client_id']);
+$payment_id = monero_payment_id($invoice_id);
```

Comment:
Could you please improve the security of monero_payment_id function? You will be able to inject potentential and dangerous scripts.

Reply:
I don't know how to improve it. :(